### ISSUE TITLE
 could not settle on kex algorithm both client and server have : curve25519-sha256@libssh.org

### ISSUE THREAD

**Issue by mariuz** [net-ssh/net-ssh#747]
### Expected behavior

When deploying/connecting to remote server using latest netssh from git (or 6.0 beta2)

### Actual behavior

Exception while executing as user@example.com: could not settle on kex algorithm (SSHKit::Runner::ExecuteError)
Server kex preferences: curve25519-sha256@libssh.org,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group16-sha512
Client kex preferences: curve25519-sha256,curve22519-sha256@libssh.org,ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group-exchange-sha256,diffie-hellman-group14-sha1

### System configuration

- net-ssh version 6.0 
- Ruby version : 2.6.5

ed25519 , bcrypt_pbkdf , x25519 
are all added to gemfile 

### Example App

When i run ssh -vv i see that open ssh is using 

server->client cipher: chacha20-poly1305@openssh.com

ssh -v

debug1: kex: algorithm: curve25519-sha256@libssh.org
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none

So this may be related 

https://github.com/net-ssh/net-ssh/issues/477

**Comment by mfazekas** [comment]
Should be fixed by https://github.com/net-ssh/net-ssh/pull/741/files
can you try with current master?!

**Comment by mariuz** [comment]
Tested , the same  

Using net-ssh 6.0.0.beta2 from git://github.com/net-ssh/net-ssh.git (at master@54476fa)

**Comment by mariuz** [comment]
ssh-audit 2.2

```
# general
(gen) banner: SSH-2.0-OpenSSH_7.4
(gen) software: OpenSSH 7.4
(gen) compatibility: OpenSSH 7.3+, Dropbear SSH 2016.73+
(gen) compression: enabled (zlib@openssh.com)

# key exchange algorithms
(kex) curve25519-sha256@libssh.org   -- [info] available since OpenSSH 6.5, Dropbear SSH 2013.62
(kex) diffie-hellman-group18-sha512  -- [info] available since OpenSSH 7.3
(kex) diffie-hellman-group14-sha256  -- [info] available since OpenSSH 7.3, Dropbear SSH 2016.73
(kex) diffie-hellman-group16-sha512  -- [info] available since OpenSSH 7.3, Dropbear SSH 2016.73

# host-key algorithms
(key) ssh-rsa (2048-bit)             -- [info] available since OpenSSH 2.5.0, Dropbear SSH 0.28
(key) rsa-sha2-512 (2048-bit)        -- [info] available since OpenSSH 7.2
(key) rsa-sha2-256 (2048-bit)        -- [info] available since OpenSSH 7.2
(key) ssh-ed25519                    -- [info] available since OpenSSH 6.5

# encryption algorithms (ciphers)
(enc) chacha20-poly1305@openssh.com  -- [info] available since OpenSSH 6.5
                                     `- [info] default cipher since OpenSSH 6.9.
(enc) aes256-gcm@openssh.com         -- [info] available since OpenSSH 6.2
(enc) aes128-gcm@openssh.com         -- [info] available since OpenSSH 6.2
(enc) aes256-ctr                     -- [info] available since OpenSSH 3.7, Dropbear SSH 0.52
(enc) aes192-ctr                     -- [info] available since OpenSSH 3.7
(enc) aes128-ctr                     -- [info] available since OpenSSH 3.7, Dropbear SSH 0.52

# message authentication code algorithms
(mac) hmac-sha2-512-etm@openssh.com  -- [info] available since OpenSSH 6.2
(mac) hmac-sha2-256-etm@openssh.com  -- [info] available since OpenSSH 6.2
(mac) umac-128-etm@openssh.com       -- [info] available since OpenSSH 6.2

# fingerprints
(fin) ssh-ed25519: SHA256:mfkHGP3NwntLNB3oOGX6C3ZsV4Gw1oItpoTL0q9XRxs
(fin) ssh-rsa: SHA256:GjKh56a2y+4ssTYZkGYTXHMZzUEwKfh1d6vnNSl9j1E

# algorithm recommendations (for OpenSSH 7.4)
(rec) +curve25519-sha256             -- kex algorithm to append 
(rec) +diffie-hellman-group-exchange-sha256-- kex algorithm to appen
```

**Comment by mfazekas** [comment]
@mariuz is the issue is still there pls report the message, with the latest version net-ssh 6.0.0 final for example.

The reported issue `curve22519-sha256@libssh.org` was fixed to  `curve25519-sha256@libssh.org`. Note the 22519 vs 25519


>Server kex preferences: curve25519-sha256@libssh.org,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group16-sha512
Client kex preferences: curve25519-sha256,curve22519-sha256@libssh.org,ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group-exchange-sha256,diffie-hellman-group14-sha1
